x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -57,12 +63,14 @@ export class AlertsClient {
   private readonly auditLogger?: AuditLogger;
   private readonly authorization: PublicMethodsOf<AlertingAuthorization>;
   private readonly esClient: ElasticsearchClient;
+  private readonly spaceId: Promise<string | undefined>;
 
   constructor({ auditLogger, authorization, logger, esClient }: ConstructorOptions) {
     this.logger = logger;
     this.authorization = authorization;
     this.esClient = esClient;
     this.auditLogger = auditLogger;
+    this.spaceId = this.authorization.getSpaceId();
```

Comment:
It looks like the rules registry already has a dependency on the spaces plugin, so would it make more sense to get the space id directly from the spaces plugin in the rules registry instead of the alerting authorization client?

Reply:
To be clear, I don't have an issue with using the auth client to get the space id if that makes the most sense but I think this alerts client already has access to the Kibana request and the spaces plugin so it might be the most straightforward to get it without relying on the auth client?

Reply:
While it is true that the rule registry will have access to the spaces plugin, in my opinion it is better to have this added as part of the authz class. In the future the space id should actually become a part of the authorization filter, and not something hard coded into the alerts as data clients "fetch" method.